server/sync.go

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2022-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/mattermost/mattermost-plugin-calls/server/cluster"
+)
+
+// lockCall locks the global (cluster) mutex for the given callID and
+// returns the current state.
+func (p *Plugin) lockCall(callID string) (*channelState, error) {
```

Comment:
Not sure there's a perfect way to clean it up. If we think we'll never decouple calls from channels then I suppose we could revert to using `channelID` in 99% of occurrences. Still we need a unique ID per call to do some consistency checks (e.g. did a new call start after the previous one?). Overall I was trying to move us in the other direction where a call doesn't necessarily have to happen in a channel but if it's confusing we can change course I suppose.

Reply:
I think having a `callID` is definitely useful (like how it's necessary for notifications). But what I meant is how the code sometimes turns a `channelID` into a `callID` (like here). That's the confusing part. If we can clean that up, that would cause less confusing. e.g., if lockCall will use a `callID`,  then only use a `callID` when calling it. Otherwise maybe call it 'lockID' or something, so that we don't think it's asking for a `callID` specifically. Hope that makes sense?

Reply:
Then this should be a `channelID`. I named it `callID` because the lock has the scope of a call. But calls can only happen in channels right now so what we really end up using is the channel ID also because it's particularly convenient that the ID exists before a call does.

I can rename this instance but if you wanted to clean it up properly we'd have to make breaking changes. Something I think we could track as a separate effort.

Reply:
> so what we really end up using is the channel ID also because it's particularly convenient that the ID exists before a call does.

Then would it be okay to rename the parameter `channelID`? That would clear up the confusion on my end.

Reply:
Sure, no concerns there :+1: